src/components/PostData/PostDetails.tsx

```diff
@@ -0,0 +1,85 @@
+import React, {
+  useCallback, useContext, useEffect, useState,
+} from 'react';
+import { NewCommentForm } from '../Comment/NewCommentForm';
+import { DispatchContext, StateContext } from '../../reducer/store';
+import { Loader } from '../Loader';
+import { CommentsList } from '../Comment/CommentsList';
+import { postService } from '../../services/post.service';
+import { Notifications } from '../Notifications/Notifications';
+
+export const PostDetails: React.FC = () => {
+  const [showSpinner, setShowSpinner] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [showCommentForm, setShowCommentForm] = useState(false);
+
+  const { selectedPost, commentsList } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const onShowFormHandler = useCallback(() => {
+    setShowCommentForm(true);
+  }, []);
+
+  const showNoComment = !isError && !commentsList && !showSpinner;
+  const showCommentsList = commentsList?.length && !showSpinner;
+  const showWriteBtn = !showCommentForm && !showSpinner && !isError;
+
+  useEffect(() => {
+    const { setPostComments } = postService(dispatch);
```

Comment:
You are calling a method from a service directly in useEffect. It's better to wrap it in a try...catch block for handling any errors.